### DISCUSSION TITLE
How to use ngrok with Warden?

### DISCUSSION THREAD

**Discussion by prince-prc** [wardenenv/warden#562] [Help & Support]
Hi,

I would like to know if it would be possible to integrate ngrok in Warden and how please.

Thanks.

**Comment by WinstonN** [comment]
Mate, you'll have more success praying for rain - see https://github.com/davidalger/warden/issues/328

But, if you want to we can go at it together, and hopefully crack it? DM me on LI

**Comment by prince-prc** [comment]
Hi @WinstonN  Thanks for your reply, I finally gave up on this idea, thanks again for your help ;)

**Comment by palejosdeuna** [comment]
Hi guys, could you share the solution?

**Comment by emreakay** [comment]
for https
ngrok http --host-header=your-warden-domain.test 443

for http
ngrok http --host-header=your-warden-domain.test 80

**Reply by navarr** [reply]
The problem I've found with ngrok is that there isn't any setting or method to rewrite URLs in the body of the page output.  That's essentially the missing feature to make this work ootb since Magento outputs the full URL for its assets and links.

**Reply by emreakay** [reply]
you may use magento's secure_url with ngrock and uncesure_url with warden's url. 
this may help you do it.

**Comment by bruno-blackbird** [comment]
Hello, 
ls someone still interested on having ngrok available in Warden ?
I implemented it in my Agency, with caddy to rewrite urls in body and headers, so you do not need to change any configuration in your Project to make it work.

**Reply by navarr** [reply]
Very much so, yes!

**Reply by conflicker** [reply]
Is there a link for this?

**Comment by beeman00** [comment]
I made it work for me!

- start a tunnel: `ngrok http  443`

```                                                                                                                                                                        
Session Status                online
Version                       3.18.1
Latency                       61ms
Web Interface                 http://127.0.0.1:4040
Forwarding                    https://c809-190-21-166-28.ngrok-free.app -> https://localhost:443
 ```


- Replace the `TRAEFIK_DOMAIN` value with the ngrock domain in your `.env` file

```
WARDEN_ENV_NAME=myenvname
WARDEN_ENV_TYPE=magento2
WARDEN_WEB_ROOT=/

TRAEFIK_DOMAIN=c809-190-21-166-28.ngrok-free.app
TRAEFIK_SUBDOMAIN=app
```
- run `warden env up` 
- change your magento base_url to the one given by ngrock

```
# via CLI to the env.php file
bin/magento config:set -e web/secure/base_url https://app.c809-190-21-166-28.ngrok-free.app/
bin/magento config:set -e web/unsecure/base_url https://app.c809-190-21-166-28.ngrok-free.app/
```

and that's how it works for me!

Also, ngrok allows you to set a reserved domain which is really helpful.
